libraries/chain/db_market.cpp

```diff
@@ -93,6 +92,78 @@ void database::globally_settle_asset( const asset_object& mia, const price& sett
 
 } FC_CAPTURE_AND_RETHROW( (mia)(settlement_price) ) }
 
+void database::revive_bitasset( const asset_object& bitasset, const account_id_type& shorter )
+{ try {
+   FC_ASSERT( bitasset.is_market_issued() );
+   const asset_bitasset_data_object& bad = bitasset.bitasset_data(*this);
+   FC_ASSERT( bad.has_settlement() );
+   const asset_dynamic_data_object& bdd = bitasset.dynamic_asset_data_id(*this);
+   FC_ASSERT( shorter != account_id_type() );
+   FC_ASSERT( !bad.is_prediction_market );
+   FC_ASSERT( !bad.current_feed.settlement_price.is_null() );
+   // use settlement_fund to create a call order, adjust supply
+   auto to_short = bdd.current_supply;
+   auto collateral = bad.settlement_fund;
+   if( collateral > 0 )
+      adjust_balance( bitasset.issuer, asset( collateral, bad.options.short_backing_asset ) );
```

Comment:
Right, this is the part where auto-revival happens after the feed has recovered. A call_order is created from debt and collateral, and this call_order belongs to the issuer.